### PR TITLE
Unit test for dapp transaction - Subtask #456 - Closes #841

### LINKED ISSUE #841: Add Unit tests for logic/dapp.js
Parent issue: #456

More unit tests are needed for this section of the code.

https://github.com/LiskHQ/lisk/blob/development/logic/dapp.js

### PR DESCRIPTION
**Pull request by SargeKhan**
Parent issue: #456 
Closes child issue:  #841

### CODE REVIEW

File: test/unit/logic/dapp.js [hunk 1]
```diff
@@ -0,0 +1,1495 @@
...
+						trs.asset.dapp.type = -1;
+
+						dapp.verify(trs, sender, function (err) {
+							expect(err).to.equal('Invalid application type');
+							done();
+						});
+					});
+				});
+
+				describe('when dapp link is not an actually link', function () {
```
Comment (Isabello) [inline test/unit/logic/dapp.js]: when dapp link is not a valid url format

File: test/unit/logic/dapp.js [hunk 2]
```diff
@@ -0,0 +1,1087 @@
...
+
+			it('should update private unconfirmed links variable', function (done) {
+				dapp.apply(transaction, dummyBlock, sender, function (err, cb) {
+					expect(unconfirmedLinks[transaction.asset.dapp.link]).to.not.exist;
+					done();
+				});
+			});
+		});
+
+		describe('undo', function () {
```
Comment (MaciejBaj) [inline test/unit/logic/dapp.js]: Please refactor the function to match current standards

File: test/unit/logic/dapp.js [hunk 3]
```diff
@@ -0,0 +1,1087 @@
...
+		});
+
+		describe('process', function () {
+
+			it('should call the callback', function (done) {
+				dapp.process(transaction, sender, done);
+			});
+		});
+
+		describe('getBytes', function () {
```
Comment (MaciejBaj) [inline test/unit/logic/dapp.js]: Please refactor the function to match current standards

File: test/unit/logic/dapp.js [hunk 4]
```diff
@@ -0,0 +1,1087 @@
...
+					dapp.verify(transaction, sender, function (err, res) {
+						expect(err).to.not.exist;
+						expect(res).to.eql(transaction);
+						done();
+					});
+				});
+			});
+		});
+
+		describe('process', function () {
```
Comment (MaciejBaj) [inline test/unit/logic/dapp.js]: Please refactor the function to match current standards

**Pull request merged**